flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/source/reader/deserializer/JodaConverter.java

```diff
@@ -0,0 +1,49 @@
+package com.google.cloud.flink.bigquery.source.reader.deserializer;
+
+import com.google.cloud.Timestamp;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Encapsulates joda optional dependency. Instantiates this class only if joda is available on the
+ * classpath.
+ *
+ * <p>This implementation is copied from
+ */
+class JodaConverter {
+
+    private static JodaConverter instance;
+    private static boolean instantiated = false;
+
+    public static JodaConverter getConverter() {
+        if (instantiated) {
+            return instance;
+        }
+
+        try {
+            Class.forName(
+                    "org.joda.time.DateTime",
+                    false,
+                    Thread.currentThread().getContextClassLoader());
+            instance = new JodaConverter();
+        } catch (ClassNotFoundException e) {
+            instance = null;
```

Comment:
Can we handle this in a better way? Why are we supplying null to runtime?

Reply:
Fixed.